release/src/main/scripts/build_release_candidate.sh

```diff
@@ -0,0 +1,220 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# This script will create a Release Candidate, includes:
+
+set -e
+
+LOCAL_CLONE_DIR=build_release_candidate
+LOCAL_JAVA_STAGING_DIR=java_staging_dir
+LOCAL_PYTHON_STAGING_DIR=python_staging_dir
+LOCAL_WEBSITE_UPDATE_DIR=website_update_dir
+LOCAL_PYTHON_DOC=python_doc
+LOCAL_JAVA_DOC=java_doc
+LOCAL_WEBSITE_REPO=beam_website_repo
+
+USER_REMOTE_URL=
+USER_GITHUB_ID=
+GIT_REPO_URL=https://github.com/apache/beam.git
+ROOT_SVN_URL=https://dist.apache.org/repos/dist/dev/beam
+GIT_BEAM_ARCHIVE=https://github.com/apache/beam/archive
+GIT_BEAM_WEBSITE=https://github.com/apache/beam-site.git
+
+PYTHON_ARTIFACTS_DIR=python
+BEAM_ROOT_DIR=beam
+WEBSITE_ROOT_DIR=beam-site
+
+
+echo "================Setting Up Environment Variables==========="
+echo "Which release version are you working on: "
+read RELEASE
+RELEASE_BRANCH=release-${RELEASE}
+echo "Which release candidate number(1-3) are you going to create: "
+read RC_NUM
+echo "Please enter your repo URL forked from apache/beam-site"
+read USER_REMOTE_URL
+echo "Please enter your github username(ID): "
+read USER_GITHUB_ID
+
+echo "================Checking Environment Variables=============="
+echo "beam repo will be cloned into: ${LOCAL_CLONE_DIR}"
+echo "working on release version: ${RELEASE}"
+echo "working on release branch: ${RELEASE_BRANCH}"
+echo "will create release candidate: RC${RC_NUM}"
+echo "Your forked beam-site URL: ${USER_REMOTE_URL}"
+echo "Please review all environment variables and confirm: [y|N]"
+read confirmation
+if [[ $confirmation != "y" ]]; then
+  echo "Please rerun this script and make sure you have the right inputs."
+  exit
+fi
+
+echo "============Building and Staging Java Artifacts============="
+echo "--------Cloning Beam Repo and Checkout Release Branch-------"
+cd ~
+mkdir ${LOCAL_CLONE_DIR}
+cd ${LOCAL_CLONE_DIR}
+git clone ${GIT_REPO_URL}
+cd ${BEAM_ROOT_DIR}
+git checkout ${RELEASE_BRANCH}
+
+echo "-------------Building Java Artifacts with Gradle-------------"
+git config credential.helper store
+
+./gradlew release -Prelease.newVersion=${RELEASE}-SNAPSHOT \
+              -Prelease.releaseVersion=${RELEASE}-RC${RC_NUM} \
+              -Prelease.useAutomaticVersion=true --info --no-daemon
+
+echo "-------------Staging Java Artifacts into Maven---------------"
+./gradlew publish -PisRelease --no-parallel --no-daemon
+echo "Please review all artifacts in https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.beam%22"
+rm -rf ~/${LOCAL_CLONE_DIR}
+
+echo "=========Staging Source Release on dist.apache.org==========="
+cd ~
+mkdir ${LOCAL_JAVA_STAGING_DIR}
+cd ${LOCAL_JAVA_STAGING_DIR}
+svn co ${ROOT_SVN_URL}
+mkdir beam/${RELEASE}
+cd beam/${RELEASE}
+
+echo "----------------Downloading Source Release-------------------"
```

Comment:
Do you know why do we create a source release by downloading from github? Is it because is it easy?

I suppose we could clone the repo and zip it.

Reply:
I don't know the reason but maybe it's easier to download directly then sign and hash it.